docusaurus/docs/Android/03-ui/04-components/02-channel-list.mdx

```diff
@@ -446,3 +446,41 @@ public sealed class EventHandlingResult {
 
 You can also use `BaseChatEventHandler` - a more detailed implementation that covers the most generic cases and forces you to implement the required ones.
 
+To inject the ChatEventHandler to `ChannelListViewModel`, use `ChatEventHandlerFactory`, inside `ChannelListViewModelFactory`: 
+
+```
+public class ChannelListViewModelFactory @JvmOverloads constructor(
+    private val filter: FilterObject? = null,
+    private val sort: QuerySort<Channel> = ChannelListViewModel.DEFAULT_SORT,
+    private val limit: Int = 30,
+    private val messageLimit: Int = 1,
+    private val chatEventHandlerFactory: ChatEventHandlerFactory = ChatEventHandlerFactory(),
+) 
+```
+
+There's a default implementation of `ChatEventHandler` for a ChannelListView listing channels that the current user is not a member. You can override the factory and inject it:
```

Comment:
```suggestion
A default `ChatEventHandlerFactory` implementation provides a `ChatEventHandler` that updates the `ChannelListView` based on membership - the channel will be added if the current user is a member and it will be removed otherwise.  You can override the factory, like in the example below:
```